Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -2230,7 +2232,29 @@ namespace AZ
                             propertyIndex = material->FindPropertyIndex(s_emissive_intensity_Name);
                             if (propertyIndex.IsValid())
                             {
-                                subMesh.m_emissiveColor *= material->GetPropertyValue<float>(propertyIndex);
+                                auto unitPropertyIndex = material->FindPropertyIndex(s_emissive_unit_Name);
+                                AZ_Assert(
+                                    propertyIndex.IsValid(),
+                                    "MeshFeatureProcessor: Materials with an emissive intensity need a unit for the intensity");
+                                auto intensity = material->GetPropertyValue<float>(propertyIndex);
+                                auto unit = material->GetPropertyValue<uint32_t>(unitPropertyIndex);
+                                bool foundEmissiveUnitFunctor = false;
+                                for (const auto& functor : material->GetAsset()->GetMaterialFunctors())
+                                {
+                                    auto emissiveFunctor = azdynamic_cast<ConvertEmissiveUnitFunctor*>(functor);
+                                    if (emissiveFunctor != nullptr)
+                                    {
+                                        intensity = emissiveFunctor->GetProcessedValue(intensity, unit);
+                                        foundEmissiveUnitFunctor = true;
+                                        break;
+                                    }
+                                }
+                                material->GetAsset()->GetId().ToFixedString().c_str();
+                                AZ_Assert(
+                                    foundEmissiveUnitFunctor,
+                                    "MeshFeatureProcessor: Could not find ConvertEmissiveUnitFunctor for material %s",
+                                    material->GetAsset()->GetId().ToFixedString().c_str());
+                                subMesh.m_emissiveColor *= intensity;
```

Comment:
Same as with the material property index, I think there should be an if block checking for the functor pointer and the assert should be a warning because it's data-driven and possible that other materials will not have the functor.